hack/cleanup/cleanup_oapi.py

```diff
@@ -0,0 +1,130 @@
+def checkImageIdExist(gateway, imageId, accountOwnerImageId):
+    """Check Image Id exist"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        if image['ImageId'] == imageId:
+            print("ImageId {} does exist".format(imageId))
+            return True
+    print("ImageId {} does not exist".format(imageId))
+    return False
+    
+# deleteImage delete image.
+def deleteImage(gateway, accountOwnerImageId, daysCount):
+    """Delete Image"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        imageId = image['ImageId']
+        creationDate = image['CreationDate']
+        print("Get ImageId {} in {}".format(imageId, creationDate))
+        checkImageToBeDelete(gateway, accountOwnerImageId, creationDate, imageId, daysCount)
+
+# deleteUnattachedImage delete unattached image
+def deleteUnattachedImage(gateway, imageId, accountOwnerImageId):
+    """Delete Unattahed Image"""
+    imageIdAttached = checkImageIdAttached(gateway, imageId, accountOwnerImageId)
+    if imageIdAttached:
+        print("Can not delete imageId {} because at least one vm use this image".format(imageId))
+    else:
+        print("Delete ImageId {}".format(imageId))
+        deleteImageId(gateway, imageId, accountOwnerImageId)
+
+# checkImageToBeDelete check if image need to be delete
+def checkImageToBeDelete(gateway, accountOwnerImageId, creationDate, imageId, daysCount):
+    """Check Image DateTime"""
+    omiAgeLimit = datetime.now() - timedelta(days=daysCount)
+    imageDate = datetime.strptime(creationDate, '%Y-%m-%dT%H:%M:%S.%fZ')
+    if imageDate < omiAgeLimit:
+        print("ImageId {} will be deleted because its is more than {} days".format(imageId, daysCount))
+        deleteUnattachedImage(gateway, imageId, accountOwnerImageId)
+    else:
+        print("ImageId {} will be saved because its is less than {} days".format(imageId, daysCount))
+
+    
+def main():
+    """Main function to delete old image
+    """
+    access_key = os.environ['OSC_ACCESS_KEY']
+    secret_key = os.environ['OSC_SECRET_KEY']
+    region = os.environ['OSC_REGION']
+
+    args = parseArgs()
+    daysCount = args.days
+    imageId = args.imageId
+    accountOwnerImageId = args.owner
+    imageName = args.imageName
+    print("Cleanup image")
+    gateway = OSCGateway(
+        **{"access_key": access_key, "secret_key": secret_key, "region": region}
+    )
+    if imageName != 'debian123':
+        imageId = getImageId(gateway, imageName, accountOwnerImageId)
+    if imageId != 'ami-00000000':
+        print("Image {} with id {} will be checked".format(imageName, imageId))
+        checkImageId = checkImageIdExist(gateway, imageId, accountOwnerImageId)
+        if checkImageId:
+            deleteUnattachedImage(gateway, imageId, accountOwnerImageId) 
+    else:
+        deleteImage(gateway, accountOwnerImageId, daysCount)
```

Comment:
Whatch out, you will delete all the image of the account older than X days ! Other OMI can be present on the account

Reply:
Ok it is done